railties/lib/rails/generators/base.rb

```diff
@@ -27,7 +27,7 @@ class Base < Thor::Group
       strict_args_position!
 
       def self.exit_on_failure? # :nodoc:
-        false
+        true
```

Comment:
This PR seems to be about exiting with a non-zero status for the _migration_ generator specifically, but this will impact all generators. Should we define this on the migration generator instead of the base generator class?